parser.go

```diff
@@ -160,7 +161,7 @@ func closeReader(reader io.ReadCloser) {
 func validateHeaders(headers http.Header) *Error {
 
 	reqContentType := headers.Get("Content-Type")
-	if reqContentType != ContentType {
+	if strings.Contains(reqContentType, ContentType) == false {
```

Comment:
Thoughts on utilizing `strings.HasPrefix` instead here? I assume we always want `application/vnd.api+json` to be the initial part of this header value?

Reply:
I was not sure if the position was defined in the specification. If it's always the first type then yes, it would be better.